specification/blueprint/resource-manager/Microsoft.Blueprint/preview/2018-11-01-preview/assignmentOperation.json

```diff
@@ -0,0 +1,322 @@
+      "get": {
+        "tags": [
+          "AssignmentOperations"
+        ],
+        "operationId": "AssignmentOperations_Get",
+        "description": "Get a Blueprint assignment operation.",
+        "x-ms-examples": {
+          "Assignment": {
+            "$ref": "./examples/BlueprintAssignmentOperation_Get.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ScopeParameter"
+          },
+          {
+            "$ref": "#/parameters/AssignmentNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AssignmentOperationNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Blueprint assignment retrieved.",
+            "schema": {
+              "$ref": "#/definitions/AssignmentOperation"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AssignmentOperation": {
+      "type": "object",
+      "description": "Represents underlying deployment detail for each update to the assignment.",
+      "allOf": [
+        { "$ref": "#/definitions/AzureResourceBase" }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Properties for AssignmentOperation",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/AssignmentOperationProperties"
+        }
+      }
+    },
+    "AssignmentOperationList":{
+      "type": "object",
+      "description": "List of AssignmentOperation.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "List of AssignmentOperation.",
+          "items": {
+            "$ref": "#/definitions/AssignmentOperation"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Link to the next page of results."
+        }
+      }
+
+    },
+    "AssignmentOperationProperties": {
+      "type": "object",
+      "description": "properties of AssignmentOperation.",
+      "properties": {
+        "blueprintVersion": {
+          "type": "string",
+          "description": "The blueprint version used for the assignment operation."
+        },
+        "assignmentState": {
+          "type": "string",
+          "description": "State of this assignment operation."
+        },
+        "timeCreated": {
+          "type": "string",
+          "description": "Create time of this Assignment Operation."
+        },
+        "timeStarted": {
+          "type": "string",
+          "description": "Start time of the underlying deployment."
+        },
+        "timeFinished": {
+          "type": "string",
+          "description": "Finish time of the overall underlying deployments."
+        },
+        "deployments": {
```

Comment:
Can this be modeled as a child action or resourceType? 
The URI would look something like:
/{scope}/providers/Microsoft.Blueprint/blueprintAssignments/{assignmentName}/operations/{assignmentOperationName}/deployments

Reply:
no, 
1. assignmentOperation is already a child resource, with result aggregation.
2. in our backend, we store AssignmentOperation as a whole unit.
3. the size of deploymentJob is pretty small, just some metadata, no huge payload.
4. it's easier to consume.